auv_sim/auv_sim_bringup/launch/start_gazebo.launch

```diff
@@ -1,49 +1,55 @@
 <?xml version="1.0"?>
 <launch>
+  <env name="GAZEBO_MODEL_PATH" value="$(find auv_sim_description)/models:${GAZEBO_MODEL_PATH}"/>
   <arg name="namespace" default="taluy" />
   <arg name="control_rate" default="20.0" />
-  <arg name="use_ned_frame" default="false" />
   <arg name="reference_frame" value="world" />
-
   <arg name="use_gui" default="true" />
   <arg name="debug" default="false" />
-  <arg name="world_name" default="$(find auv_sim_description)/worlds/seabed.world" />
-  <arg name="velocity_control" default="true" />
-  <arg name="joy_id" default="0" />
+  <arg name="world" default="pool" />
+  <arg name="use_ned_frame" default="false" />
+
+    <!--Run rviz-->
+  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find taluy_description)/rviz/yolo_cam.rviz"
+    if="$(arg use_gui)" />
+
 
-  <arg name="x" value="9.0" />
-  <arg name="y" value="-3.0" />
-  <arg name="z" value="-87.0" />
+
+  <!--Initial robot poisition and orientation-->
+  <arg name="x" value="0.0" />
+  <arg name="y" value="0.0" />
+  <arg name="z" value="-1.50" />
+  <arg name="z_sea" value="-87" if="$(eval arg('world') == 'seabed')" />
```

Comment:
I think this is rather a complicated logic here. Maybe use:

```xml

  <group if="$(eval arg('world') == 'seabed')">
    <arg name="x" value="0.0" />
    <arg name="y" value="0.0" />
    <arg name="z" value="-87.0" />
  </group>


 <group if="$(eval arg('world') == 'pool')">
    <arg name="x" value="0.0" />
    <arg name="y" value="0.0" />
    <arg name="z" value="-1.5" />
  </group>
```

Though I'm not sure if this is the case for launch arguments.

Reply:
When using a group, its arguments apply only within that group. So your recommendation causes the error: RLException: arg 'x' is not defined.

One could first define the arguments and then try to override in groups:
```xml
<!-- Define default arguments outside groups -->
<arg name="x" default="0.0" />
<arg name="y" default="0.0" />
<arg name="z" default="-1.5" />

<!-- Override defaults based on world -->
<group if="$(eval arg('world') == 'seabed')">
    <arg name="x" value="$(arg x)" />
    <arg name="y" value="$(arg y)" />
    <arg name="z" value="-87.0" />
</group>

<group if="$(eval arg('world') == 'pool')">
    <arg name="x" value="$(arg x)" />
    <arg name="y" value="$(arg y)" />
    <arg name="z" value="-1.5" />
</group>
```
But now this gives an override error:
RLException: Invalid <arg> tag: arg 'x' has already been declared. 

So there doesnt seem to be an easy way to group like your suggestion. However, I think this new approach is good and flexible enough (at least for now):
```xml
    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="-1.5" if="$(eval arg('world') == 'pool')" />
    <arg name="z" default="-87.0" if="$(eval arg('world') == 'seabed')" />
    <arg name="roll" value="0" />
    <arg name="pitch" value="0" />
    <arg name="yaw" value="-1.8" />
```
and then simply
```xml
<arg name="z" value="$(arg z)" />
```